Compile error with ureq v3

After  `cargo update`, we get a compile error with ureq 3.

```
error[E0599]: no method named `set` found for struct `RequestBuilder` in the current scope
  --> src/options/update.rs:22:18
   |
21 | /             ureq::get("https://api.github.com/repos/Yamato-Security/hayabusa/releases/latest")
22 | |                 .set("User-Agent", "HayabusaUpdateChecker")
   | |                 -^^^ method not found in `RequestBuilder<WithoutBody>`
   | |_________________|
   |

For more information about this error, try `rustc --explain E0599`.
```

@fukusuket Could you look at this when you get a chance?

--- a/src/main.rs
+++ b/src/main.rs
@@ -608,7 +608,7 @@ impl App {
                     match get(url).call() {
                         Ok(mut res) => {
                             let mut dst = File::create(Path::new("./encoded_rules.yml")).unwrap();
-                            dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes())
+                            dst.write_all(res.body_mut().read_to_vec().unwrap().as_bytes())
                                 .unwrap();
                             write_color_buffer(
                                 &BufferWriter::stdout(ColorChoice::Always),

--- a/src/main.rs
+++ b/src/main.rs
@@ -608,7 +608,8 @@ impl App {
                     match get(url).call() {
                         Ok(mut res) => {
                             let mut dst = File::create(Path::new("./encoded_rules.yml")).unwrap();
-                            dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes()).unwrap();
+                            dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes())
+                                .unwrap();
                             write_color_buffer(
                                 &BufferWriter::stdout(ColorChoice::Always),
                                 get_writable_color(
@@ -631,7 +632,8 @@ impl App {
                             Ok(mut res) => {
                                 let mut dst =
                                     File::create(Path::new("./rules_config_files.txt")).unwrap();
-                                dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes()).unwrap();
+                                dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes())
+                                    .unwrap();
                                 write_color_buffer(
                                     &BufferWriter::stdout(ColorChoice::Always),
                                     get_writable_color(

--- a/src/main.rs
+++ b/src/main.rs
@@ -608,7 +608,7 @@ impl App {
                     match get(url).call() {
                         Ok(mut res) => {
                             let mut dst = File::create(Path::new("./encoded_rules.yml")).unwrap();
-                            dst.write_all(res.body_mut().read_to_vec().unwrap().as_bytes())
+                            dst.write_all(res.body_mut().read_to_vec().unwrap().as_slice())
                                 .unwrap();
                             write_color_buffer(
                                 &BufferWriter::stdout(ColorChoice::Always),

--- a/src/main.rs
+++ b/src/main.rs
@@ -8,7 +8,7 @@ use std::borrow::BorrowMut;
 use std::ffi::{OsStr, OsString};
 use std::fmt::Display;
 use std::fmt::Write as _;
-use std::io::{copy, BufRead, BufWriter, Write};
+use std::io::{BufRead, BufWriter, Write};
 use std::path::Path;
 use std::ptr::null_mut;
 use std::sync::Arc;
@@ -606,9 +606,9 @@ impl App {
                 if rule_encoded.exists() {
                     let url = "https://raw.githubusercontent.com/Yamato-Security/hayabusa-encoded-rules/main/encoded_rules.yml";
                     match get(url).call() {
-                        Ok(res) => {
+                        Ok(mut res) => {
                             let mut dst = File::create(Path::new("./encoded_rules.yml")).unwrap();
-                            copy(&mut res.into_reader(), &mut dst).unwrap();
+                            dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes()).unwrap();
                             write_color_buffer(
                                 &BufferWriter::stdout(ColorChoice::Always),
                                 get_writable_color(
@@ -628,10 +628,10 @@ impl App {
                     if !ONE_CONFIG_MAP.is_empty() {
                         let url = "https://raw.githubusercontent.com/Yamato-Security/hayabusa-encoded-rules/refs/heads/main/rules_config_files.txt";
                         match get(url).call() {
-                            Ok(res) => {
+                            Ok(mut res) => {
                                 let mut dst =
                                     File::create(Path::new("./rules_config_files.txt")).unwrap();
-                                copy(&mut res.into_reader(), &mut dst).unwrap();
+                                dst.write_all(res.body_mut().read_to_string().unwrap().as_bytes()).unwrap();
                                 write_color_buffer(
                                     &BufferWriter::stdout(ColorChoice::Always),
                                     get_writable_color(

--- a/src/options/update.rs
+++ b/src/options/update.rs
@@ -19,10 +19,11 @@ impl Update {
     pub fn get_latest_hayabusa_version() -> Result<Option<String>, Box<dyn std::error::Error>> {
         let text =
             ureq::get("https://api.github.com/repos/Yamato-Security/hayabusa/releases/latest")
-                .set("User-Agent", "HayabusaUpdateChecker")
-                .set("Accept", "application/vnd.github.v3+json")
+                .header("User-Agent", "HayabusaUpdateChecker")
+                .header("Accept", "application/vnd.github.v3+json")
                 .call()?
-                .into_string()?;
+                .body_mut()
+                .read_to_string()?;
         let json_res: Value = serde_json::from_str(&text)?;
 
         if json_res["tag_name"].is_null() {

